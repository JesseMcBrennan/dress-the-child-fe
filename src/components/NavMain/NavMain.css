.navMain {
  width: 15vw;
  height: 100vh;
  border-right: 1px solid grey;
  margin: 0; }

.navLogo {
  background: no-repeat center/10rem url("../../images/DressTheChild_noPants-01.svg");
  height: 10rem;
  width: 100%; }

img {
  height: 5em; }

.NavMain-buttons {
  display: flex;
  flex-direction: column;
  text-align: left;
  padding-inline-start: 0; }

li {
  list-style-type: none;
  padding: 10px 0 10px 0; }

.NavMain-a {
  text-decoration: none;
  padding-left: 30px;
  color: grey; }

.NavMain-a:hover {
  border-left: 2px black solid;
  padding-left: 28px; }

.Nav-item {
  width: 20rem; }
  padding-left: 35px; }

.donate:hover {
  border-left: 2px #ed1b23 solid;
  padding-left: 35px; }

.about:hover {
  border-left: 2px #00aeef solid;
  padding-left: 35px; }

.mission:hover {
  border-left: 2px #7e3e97 solid;
  padding-left: 35px; }

.partners:hover {
  border-left: 2px #e40088 solid;
  padding-left: 35px; }
