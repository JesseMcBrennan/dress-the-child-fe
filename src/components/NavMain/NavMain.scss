.navMain {
  width: 100%;
  height: 10vh;
  display: flex;
  flex-direction: row;
  justify-content: right;
}

.navLogo {
  background: no-repeat center/10rem
    url('../../images/DressTheChild_noPants-01.svg');
  height: 10rem;
  width: 100%;
}

img {
  height: 5em;
}

.NavMain-buttons {
  display: flex;
  flex-direction: row;
  text-align: right;
  width: 100vw;
  padding: 0;
}

.NavMain-a {
  text-decoration: none;
  color: whitesmoke;
}

.NavMain-a:hover {
  border-left: 2px black solid;
  padding-left: 28px;
}

.Nav-item {
  width: 10rem;
  text-align: center;
}

.donate:hover {
  border-left: 2px rgb(237, 27, 35) solid;
  padding-left: 35px;
}

.about:hover {
  border-left: 2px rgb(0, 174, 239) solid;
  padding-left: 35px;
}

.mission:hover {
  border-left: 2px rgb(126, 62, 151) solid;
  padding-left: 35px;
}

.partners:hover {
  border-left: 2px rgb(228, 0, 136) solid;
  padding-left: 35px;
}
