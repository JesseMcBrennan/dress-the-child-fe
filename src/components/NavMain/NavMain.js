import React from 'react';
import './NavMain.css';
import { BrowserRouter as Router, withRouter, NavLink } from 'react-router-dom';

export const NavMain = () => {
  return (
    <div className="navMain">
      <NavLink exact to="/">
        <div className="navLogo" />
      </NavLink>
      <ul className="NavMain-buttons">
        <li className="Nav-item">
          <NavLink exact to="/donate" className="NavMain-a">
        <li>
          <NavLink exact to="/donate" className="NavMain-a donate">
            DONATE
          </NavLink>
        </li>
        <li className="Nav-item">
          <NavLink exact to="/mission" className="NavMain-a">
            OUR MISSION
          </NavLink>
        </li>
        <li className="Nav-item">
          <NavLink exact to="/events" className="NavMain-a">
            EVENTS
          </NavLink>
        </li>
        <li className="Nav-item">
          <NavLink exact to="/volunteer" className="NavMain-a">
            VOLUNTEER
          </NavLink>
        </li>

        <li>
          <NavLink exact to="/partners" className="NavMain-a partners">
            PARTNERS
          </NavLink>
        </li>
      </ul>
    </div>
  );
};

export default withRouter(NavMain);
