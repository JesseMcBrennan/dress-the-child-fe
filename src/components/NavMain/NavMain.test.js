import React from 'react';
import { shallow } from 'enzyme';
import { NavMain } from './NavMain';

describe('NavMain', () => {
  let wrapper;

  it('should match the snapshot', () => {
    wrapper = shallow(<NavMain />);
    expect(wrapper).toMatchSnapshot();
  });
});
