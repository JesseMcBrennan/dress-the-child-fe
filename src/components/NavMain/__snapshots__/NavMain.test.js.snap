// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMain should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavMain />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          exact={true}
          to="/"
        >
          <div
            className="navLogo"
          />
        </NavLink>,
        <ul
          className="NavMain-buttons"
        >
          <li
            className="Nav-item"
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="NavMain-a"
              exact={true}
              to="/donate"
            >
              DONATE
            </NavLink>
          </li>
          <li
            className="Nav-item"
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="NavMain-a"
              exact={true}
              to="/mission"
            >
              OUR MISSION
            </NavLink>
          </li>
          <li
            className="Nav-item"
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="NavMain-a"
              exact={true}
              to="/events"
            >
              EVENTS
            </NavLink>
          </li>
          <li
            className="Nav-item"
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="NavMain-a"
              exact={true}
              to="/volunteer"
            >
              VOLUNTEER
            </NavLink>
          </li>
        </ul>,
      ],
      "className": "navMain",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <div
            className="navLogo"
          />,
          "exact": true,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "navLogo",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="Nav-item"
            >
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="NavMain-a"
                exact={true}
                to="/donate"
              >
                DONATE
              </NavLink>
            </li>,
            <li
              className="Nav-item"
            >
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="NavMain-a"
                exact={true}
                to="/mission"
              >
                OUR MISSION
              </NavLink>
            </li>,
            <li
              className="Nav-item"
            >
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="NavMain-a"
                exact={true}
                to="/events"
              >
                EVENTS
              </NavLink>
            </li>,
            <li
              className="Nav-item"
            >
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="NavMain-a"
                exact={true}
                to="/volunteer"
              >
                VOLUNTEER
              </NavLink>
            </li>,
          ],
          "className": "NavMain-buttons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="NavMain-a"
                exact={true}
                to="/donate"
              >
                DONATE
              </NavLink>,
              "className": "Nav-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "DONATE",
                "className": "NavMain-a",
                "exact": true,
                "to": "/donate",
              },
              "ref": null,
              "rendered": "DONATE",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="NavMain-a"
                exact={true}
                to="/mission"
              >
                OUR MISSION
              </NavLink>,
              "className": "Nav-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "OUR MISSION",
                "className": "NavMain-a",
                "exact": true,
                "to": "/mission",
              },
              "ref": null,
              "rendered": "OUR MISSION",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="NavMain-a"
                exact={true}
                to="/events"
              >
                EVENTS
              </NavLink>,
              "className": "Nav-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "EVENTS",
                "className": "NavMain-a",
                "exact": true,
                "to": "/events",
              },
              "ref": null,
              "rendered": "EVENTS",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="NavMain-a"
                exact={true}
                to="/volunteer"
              >
                VOLUNTEER
              </NavLink>,
              "className": "Nav-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "VOLUNTEER",
                "className": "NavMain-a",
                "exact": true,
                "to": "/volunteer",
              },
              "ref": null,
              "rendered": "VOLUNTEER",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            exact={true}
            to="/"
          >
            <div
              className="navLogo"
            />
          </NavLink>,
          <ul
            className="NavMain-buttons"
          >
            <li
              className="Nav-item"
            >
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="NavMain-a"
                exact={true}
                to="/donate"
              >
                DONATE
              </NavLink>
            </li>
            <li
              className="Nav-item"
            >
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="NavMain-a"
                exact={true}
                to="/mission"
              >
                OUR MISSION
              </NavLink>
            </li>
            <li
              className="Nav-item"
            >
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="NavMain-a"
                exact={true}
                to="/events"
              >
                EVENTS
              </NavLink>
            </li>
            <li
              className="Nav-item"
            >
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="NavMain-a"
                exact={true}
                to="/volunteer"
              >
                VOLUNTEER
              </NavLink>
            </li>
          </ul>,
        ],
        "className": "navMain",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <div
              className="navLogo"
            />,
            "exact": true,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "navLogo",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="Nav-item"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="NavMain-a"
                  exact={true}
                  to="/donate"
                >
                  DONATE
                </NavLink>
              </li>,
              <li
                className="Nav-item"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="NavMain-a"
                  exact={true}
                  to="/mission"
                >
                  OUR MISSION
                </NavLink>
              </li>,
              <li
                className="Nav-item"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="NavMain-a"
                  exact={true}
                  to="/events"
                >
                  EVENTS
                </NavLink>
              </li>,
              <li
                className="Nav-item"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="NavMain-a"
                  exact={true}
                  to="/volunteer"
                >
                  VOLUNTEER
                </NavLink>
              </li>,
            ],
            "className": "NavMain-buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="NavMain-a"
                  exact={true}
                  to="/donate"
                >
                  DONATE
                </NavLink>,
                "className": "Nav-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "DONATE",
                  "className": "NavMain-a",
                  "exact": true,
                  "to": "/donate",
                },
                "ref": null,
                "rendered": "DONATE",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="NavMain-a"
                  exact={true}
                  to="/mission"
                >
                  OUR MISSION
                </NavLink>,
                "className": "Nav-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "OUR MISSION",
                  "className": "NavMain-a",
                  "exact": true,
                  "to": "/mission",
                },
                "ref": null,
                "rendered": "OUR MISSION",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="NavMain-a"
                  exact={true}
                  to="/events"
                >
                  EVENTS
                </NavLink>,
                "className": "Nav-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "EVENTS",
                  "className": "NavMain-a",
                  "exact": true,
                  "to": "/events",
                },
                "ref": null,
                "rendered": "EVENTS",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="NavMain-a"
                  exact={true}
                  to="/volunteer"
                >
                  VOLUNTEER
                </NavLink>,
                "className": "Nav-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "VOLUNTEER",
                  "className": "NavMain-a",
                  "exact": true,
                  "to": "/volunteer",
                },
                "ref": null,
                "rendered": "VOLUNTEER",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
